src/raftstore/store/util.rs

```diff
@@ -519,8 +538,38 @@ impl Lease {
     }
 
     pub fn expire(&mut self) {
+        self.disconnect();
         self.bound = None;
     }
+
+    pub fn disconnect(&mut self) {
+        // Expire remote lease if there is any.
+        if let Some(r) = self.remote.take() {
+            r.expire();
+        }
+    }
+
+    pub fn remote(&mut self, term: u64) -> Option<RemoteLease> {
+        if self.remote.is_some() {
+            // At most one connected RemoteLease.
+            return None;
```

Comment:
Is it possible that `self.remote.term < term`?

Reply:
It is not possible, at that time, `self.remote` is `None`.